src/dcp/dcp.c

```diff
@@ -442,77 +445,139 @@ int main(int argc, char** argv)
+    if (is_posix_copy) {
+        int valid, copy_into_dir;
+        mfu_param_path_check_copy(numpaths_src, paths, destpath, mfu_src_file, mfu_dst_file,
+                                  mfu_copy_opts->no_dereference, &valid, &copy_into_dir);
+        mfu_copy_opts->copy_into_dir = copy_into_dir;
+
+        /* exit job if we found a problem */
+        if (!valid) {
+            if(rank == 0) {
+                MFU_LOG(MFU_LOG_ERR, "Invalid src/dest paths provided. Exiting run: "
+                        MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+            }
+            mfu_param_path_free_all(numpaths, paths);
+            mfu_free(&paths);
+            mfu_finalize();
+            MPI_Finalize();
+            return 1;
+        }
 
-    /* exit job if we found a problem */
-    if (!valid) {
-        if(rank == 0) {
-            MFU_LOG(MFU_LOG_ERR, "Invalid src/dest paths provided. Exiting run: "
-                    MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+        /* perform POSIX copy */
+        if (inputname == NULL) {
+            /* if daos is set to SRC then use daos_ functions on walk */
+            mfu_flist_walk_param_paths(numpaths_src, paths, walk_opts, flist, mfu_src_file);
+        } else {
+            struct mfu_flist_skip_args skip_args;
+
+            /* otherwise, read list of files from input, but then stat each one */
+            mfu_flist input_flist = mfu_flist_new();
+            mfu_flist_read_cache(inputname, input_flist);
+
+            skip_args.numpaths = numpaths_src;
+            skip_args.paths = paths;
+            mfu_flist_stat(input_flist, flist, input_flist_skip, (void *)&skip_args,
+                           walk_opts->dereference, mfu_src_file);
+            mfu_flist_free(&input_flist);
+        }
+        /* copy flist into destination */ 
+        rc = mfu_flist_copy(flist, numpaths_src, paths,
+                                destpath, mfu_copy_opts, mfu_src_file,
+                                mfu_dst_file);
+        if (rc < 0) {
+            /* hit some sort of error during copy */
+            rc = 1;
         }
+        /* free the file list */
+        mfu_flist_free(&flist);
+
+        /* free the path parameters */
         mfu_param_path_free_all(numpaths, paths);
+
+        /* free memory allocated to hold params */
         mfu_free(&paths);
-        mfu_finalize();
-        MPI_Finalize();
-        return 1;
-    }
 
-    /* create an empty file list */
-    mfu_flist flist = mfu_flist_new();
+        /* free the input file name */
+        mfu_free(&inputname);
+    } 
+#ifdef DAOS_SUPPORT
+    else {
+        daos_epoch_t epoch;
+        if (rank == 0) {
+            rc = daos_obj_list_oids(daos_args, &epoch, flist);
+	    if (rc != 0) {
+                MFU_LOG(MFU_LOG_ERR, "DAOS failed to list oids: ", MFU_ERRF,
+                        MFU_ERRP(-MFU_ERR_DAOS));
+                rc = 1;
+                
+            }
+        }
```

Comment:
looks like some indentation mismatch here, and an extra empty line we could drop

Reply:
@adammoody I keep forgetting to switch my vim configuration to not use tabs. DAOS only uses tabs, and going back and forth can be a pain :-)

Reply:
Oh, right.  Yeah, that's annoying.  Sorry about that.  We could look at the astyle options to automate the conversion if you have to copy in a bunch of code.